src/adapter.js

```diff
@@ -60,13 +60,24 @@ function getRelevantStackFrom (stack) {
  * @return {String}      Formatted step.
  */
 function formatFailedStep (step) {
-  // Safari seems to have no stack trace,
-  // so we just return the error message:
-  if (!step.stack) { return step.message }
-
   var relevantMessage = []
   var relevantStack = []
 
+  // Safari/Firefox seems to have no stack trace,
+  // so we just return the error message and if available
+  // construct a stacktrace out of filename and lineno:
+  if (!step.stack) {
+    if (step.filename) {
+      let stack = step.filename;
```

Comment:
Let's call this local 'frame' or 'stackframe' rather than 'stack'  to clarify the code.